benchbuild/environments/domain/model.py

```diff
@@ -123,20 +149,37 @@ class Image:
     name: str = attr.ib()
     from_: FromLayer = attr.ib()
     layers: tp.List[Layer] = attr.ib()
-    events = attr.ib(attr.Factory(list))  # type: tp.List[events.Event]
-    env = attr.ib(attr.Factory(dict))  # type: tp.Dict[str, str]
+    events: tp.List[Message] = attr.ib(attr.Factory(list))
+    env: tp.Dict[str, str] = attr.ib(attr.Factory(dict))
     mounts: tp.List[Mount] = attr.ib(attr.Factory(list))
+    layer_index: tp.Dict[Layer, LayerState] = attr.ib(attr.Factory(dict))
 
     def update_env(self, **kwargs: str) -> None:
         self.env.update(kwargs)
 
-    def append(self, layer: Layer) -> None:
-        self.layers.append(layer)
+    def append(self, *layers: Layer) -> None:
+        for layer in layers:
+            self.layers.append(layer)
+            self.layer_index[layer] = LayerState.ABSENT
```

Comment:
Seems like `attrs` can't generate proper `__hash__()` functions for all layers:

```
ERROR:benchbuild.environments.service_layer.messagebus:Exception handling command CreateLayer(name='localhost/debian:10_varats', container_id='debian-working-container', layer=CopyLayer(sources=['/tmp/tmp75doa268/.varats.yaml'], destination='/varats_root/.varats.yaml'))
Traceback (most recent call last):
  File "/home/sebi/.venvs/varats/lib/python3.9/site-packages/benchbuild/environments/service_layer/messagebus.py", line 83, in _handle_command
    queue.extend(handler(command))
  File "/home/sebi/.venvs/varats/lib/python3.9/site-packages/benchbuild/environments/service_layer/handlers.py", line 27, in wrapped_handler
    handler(uow, msg)
  File "/home/sebi/.venvs/varats/lib/python3.9/site-packages/benchbuild/environments/service_layer/handlers.py", line 76, in create_layer
    image.append(cmd.layer)
  File "/home/sebi/.venvs/varats/lib/python3.9/site-packages/benchbuild/environments/domain/model.py", line 163, in append
    self.layer_index[layer] = LayerState.ABSENT
  File "<attrs generated hash benchbuild.environments.domain.model.CopyLayer>", line 2, in __hash__
    return hash((
TypeError: unhashable type: 'list'

```

Reply:
@simbuerg This error is currently still on master